### ISSUE TITLE
Multiple issues with `$validate()` and `$reset()`

### ISSUE THREAD

**Issue by martinszeltins** [victorgarciaesgi/regle#19]
I noticed multiple issues with `$validate()` and `$reset()` method. I will attach a screencast so it is easy to see.

I have a form with both client and server validations.

1. When I first call `$validate()` then I see that all rules are triggered properly and all error messages appear ✅ 
2. Then I call `$reset()` and I expected all validations to reset and all error messages to disappear. _But something strange happened._ Only the server error disappeared. All other error messages stayed.  They should all go away. 🐞 
3. Then I called `$validate()` again and then I clicked on `$reset()` and then ALL error messages disappeared !! 😱
4. Then I clicked on "Save" (`$validate()`) again and all error messages appeared except the server error messages... 😕 

Seems like a lot of strange things are happening.

**Reproduction:**
[stackblitz.com/~/github.com/martinszeltins/regle-test-9](https://stackblitz.com/~/github.com/martinszeltins/regle-test-9)

**Screencast:**

https://github.com/user-attachments/assets/713e5af0-e72f-45c1-a7b9-1c9fa12a1862

**Comment by victorgarciaesgi** [comment]
That's indeed weird! I'll look

**Comment by victorgarciaesgi** [comment]
There was indeed a bug on reset with externalErrors, fixing and adding the case to the tests

**Comment by victorgarciaesgi** [comment]
So I followed this behaviour: by default, touching a field clears it's external errors (at the source).
- If you have external errors from the start, they will be cleared in the first `$validate()`, because it also touch the fields.
- But that's not a problem, because server errors should be set after a async call, after the `$validate`
- Then errors are shown
- `$reset` is called, external errors are cleared

I fixed the edge case that you had, and added tests for it. Try `0.2.9` 👍
